### ISSUE TITLE
A few comments

### ISSUE THREAD

**Issue by sauljaffe** [HarmlessHarm/HarmlessKey#32]
I just signed up for an account to test out harmlesskey to see if it's the right tool for me.  I have a number comments on first impression:

1.  I like the dark background BUT there are times when lettering is in a light grey on the dark background which makes it impossible for me to read.  Maybe people with younger eyes can read this but I can't.  The best example is on the dice roller page where the die type is in greyed lettering.  I could just make out "d4" and "d6" and the others if I put my eyes right up against the glass and squint.  Why is this not in white lettering which would stand out clearly against the dark background?

2.  I was reading through the documentation (or trying to) that you provide.  It's obvious that English is not your first language.  There are many grammatical and spelling mistakes in the documentation.  If you're going to provide documentation in English you should have someone knowledgeable in English at least proof-read it.  I won't list all the errors I found here because it would take too long.

3.  I know you want to limit those with free accounts but it's not clear exactly what you mean by some of the limitations.  For example, you say you limit the number of NPCs.  Is that per encounter?  Per account?  Per campaign?  Does it mean individual NPCs or individual groups.  For example, I think the limit is 3 NPCs for free accounts.  If I create an encounter with 4 skeletons, does that count as being over my limit?  Or are the 4 skeletons counted as 1 NPC for purposes of the encounter?  And what happens when I create a second encounter with 3 zombies?  Am I now over my limit or not?

4.  You also limit characters.  There doesn't seem to be any way to delete characters so that I cannot create a bunch of dummy characters to try things out with, and then delete them when I start using your system for my campaign with my real characters.

5.  So much more is needed during combat than name, AC and hit points.  There's ability modifiers for saves, there's spell save DCs, there's the type of weapon used, there's proficiencies, there's damage types for the weapons and a whole lot more.  Is all of this going to be added?  If not, then you are far, far, far behind many of the totally free apps that are out there and even further behind roll20 or fantasygrounds.

In short, I still don't know if I'm going to be using harmlesskeys for my campaign that I am starting tomorrow because I cannot test it.  I don't know if I can delete characters I create today for testing purposes, I can't read some of the lettering on the pages, and the documentation is awful.  It's a good idea but I don't know how well it works without testing.

Thank you.

**Comment by Lakeyzer** [comment]
Hey Sauljaffe,
Thanks for your feedback, I'll try to give a satisfying answer to all of your comments below.

1. This is something we haven't heard before, but an easy fix. Since you pointed out where the text is too dark, it's easy for us to change this color, we might even have time to do something about this tonight.

2. The documentation is quite old and quickly written. Please understand that we work on this app in our free time, we both have a job and go to university. There is just so much work we never get around to the documentation. We’d rather build a new feature or fix issues. Maybe it’s a good idea to take it down completely until we have a better documentation.

3. All the limitations are per account except for the encounters, those are per campaign. The limit for NPC’s is for the custom NPC’s that you can build there is no limit to how many NPC’s you can add to an encounter.

4. You can delete any character you have created. If you go to harmlesskey.com/players while you are logged in, you see the overview of all the players you created. Click on the trashcan to delete the character. You can create and test all you want and delete all of it later, so you are not limited in this at all.

5. The name AC and hit points are the essentials needed to run a character in our app. Of course a lot more is needed to run good encounters. Players are not where the focus of our app lies. For monsters you can create entire monster cards, with abilities you can roll, but players usually want to roll their own abilities and our app won’t take that away from them.

I hope I gave you some useful answers to your feedback. And I hope you will give our app another chance. As I said, you can create and test all you want. Everything you created can be deleted. Thanks again for your feedback!

Harm & Key

**Comment by sauljaffe** [comment]
> On Aug 26, 2019, at 4:09 PM, Lakeyzer <notifications@github.com> wrote:
> 
> Hey Sauljaffe,
> Thanks for your feedback, I'll try to give a satisfying answer to all of your comments below.
> 
> 	• This is something we haven't heard before, but an easy fix. Since you pointed out where the text is too dark, it's easy for us to change this color, we might even have time to do something about this tonight.

Not too dark - too light.

> 	• The documentation is quite old and quickly written. Please understand that we work on this app in our free time, we both have a job and go to university. There is just so much work we never get around to the documentation. We’d rather build a new feature or fix issues. Maybe it’s a good idea to take it down completely until we have a better documentation.

And then people like me will quickly just give up because having bad documentation is better than NO documentation.  People need to know how to use the system and that’s JUST AS IMPORTANT as new features and bug fixes.  Anyone who has been in the computer industry for any length of time (and I’ve been in it for about 40 years) will tell you that documentation is MANDATORY.  The more complex the system, the greater the need for documentation.

> 	• All the limitations are per account except for the encounters, those are per campaign. The limit for NPC’s is for the custom NPC’s that you can build there is no limit to how many NPC’s you can add to an encounter.

So there are a limited number of encounters I can have for my campaign?

Nope.  Sorry.  I’ll go use one of a dozen or so other sites where I can have as many encounters as I want.  For free.

> 	• You can delete any character you have created. If you go to harmlesskey.com/players while you are logged in, you see the overview of all the players you created. Click on the trashcan to delete the character. You can create and test all you want and delete all of it later, so you are not limited in this at all.

Ok.

> 	• The name AC and hit points are the essentials needed to run a character in our app. Of course a lot more is needed to run good encounters. Players are not where the focus of our app lies. For monsters you can create entire monster cards, with abilities you can roll, but players usually want to roll their own abilities and our app won’t take that away from them.

You’ve obviously either never run encounters as a DM.  Other information is essential to running an encounter.  I’ve been playing and DM’ing since original D&D and have played (and run) a lot more systems than just D&D.  How about you?

> I hope I gave you some useful answers to your feedback. And I hope you will give our app another chance. As I said, you can create and test all you want. Everything you created can be deleted. Thanks again for your feedback!

So, I can test all I want but there’s a limit to the number of encounters I can have per campaign, and a limit to the number of campaigns?  So all the tests I do count against those limits?

No.

Don’t bother to respond.  Your message won’t be read.

--
Saul Jaffe
sauljaffe@newsguy.com

**Comment by HarmlessHarm** [comment]
Since this is public, we will respond to make sure there is no confusion for others.
First of all, it saddens us to hear that you were not satisfied with our app, but we wish you all the best in your upcoming campaigns!

Now to clear up some confusion. 
It seems your main point of confusion is with the limits we have set for free accounts. You are indeed limited in the amount of content you can create. However, anything you delete frees up space for new content. For example, if you would create 3 encounters which you use during a session, you can later delete them which frees up space, allowing you to create new encounters again. This is how all our content limits work. With this principle we have tried to make it possible for anyone to run a session without paying anything, as long as  they are willing to manage their content. Paying just improves your quality of life, as you will have to manage your data a lot less. 
We believe our free options are quite fair, but of course you are entitled to your opinion.

You also mention that you find our documentation lacking. Although we have never heard anyone complain, it would surely be a good idea to take a new look at it. However, it won't be high on our priority list. 

We appreciate your feedback on the readability of some of the areas on the platform, this is something we will take a look at.

That being said, we would like to address your attitude. This platform (github) is meant to leave feedback about technical issues or make feature requests. We would like to keep it respectful here. In our opinion we have given you a friendly and respectful response to your feedback and we have trouble understanding your rather aggressive response. We welcome feedback, as it helps us immensely to improve our product, we do not however appreciate the attitude in your reply. 

We do not force you to use our app, nor do we force you to pay for it, so we really don’t understand where your frustration is coming from. We wish you the best and we hope you will find what you are looking for, whatever that may be.
